CARDS-2317 / CARDS-2320: Promote variants to a core CARDS module

sdumitriu opened the pull request:
This can be tested in the base platform with `./start_cards.sh -f mvn:io.uhndata.cards/cards-variants/0.9.22-SNAPSHOT/slingosgifeature`.

sdumitriu committed:
CARDS-2317 / CARDS-2320: Promote variants to a core CARDS module

marta- commented:
I'm starting to wonder if we've thought this part through. The functionality of variants is highly dependent on the existence of the Patient/Tumor/TumorRegion subject hierarchy, which at this time is only available in lfs.

marta- commented:
> I'm starting to wonder if we've thought this part through. The functionality of variants is highly dependent on the existence of the Patient/Tumor/TumorRegion subject hierarchy, which at this time is only available in lfs.

Nevermind, my knowledge of the subject types organization was deprecated. all good.

marta- reviewed (approved):
Looks fine, **did not test yet**. LFS should be tested to make sure it works as before. Also we should test the activation of the `variants` module in some other random project like `proms` or `heracles`.

marta- commented on modules/variants/pom.xml:
```diff
@@ -47,4 +47,25 @@
     </plugins>
   </build>
+
+  <dependencies>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>cards-patient-subject-type</artifactId>
+      <version>${project.version}</version>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>cards-tumor-subject-type</artifactId>
+      <version>${project.version}</version>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>cards-tumor-region-subject-type</artifactId>
+      <version>${project.version}</version>
+      <scope>runtime</scope>
+    </dependency>
+  </dependencies>
```
Either I'm missing something or this is not enough and we need a feature.json with these modules specified as dependencies.

I tried to turn on `variants` in `prems` by adding it as a dep in `clinical-data/pom.xml` and in `feature.json`. The sidebar entry and variants upload UI were there, but the Tumor and TumorRegion subject types were not included so variants were non-functional. We shouldn't have to add these subject types to the project manually when specifying the dependencies, they should come out of the box with `variants`.

sdumitriu commented:
You are right, a feature file would be great, now that it's easy to enable them in a project

sdumitriu committed:
CARDS-2317 / CARDS-2320: Promote variants to a core CARDS module
Added a feature file for this module to make it easier to integrate in projects

marta- commented:
> You are right, a feature file would be great, now that it's easy to enable them in a project

I'm encountering the same problem as before the icommit that introduces the variants feature file: Subject types Tumor, TumorRegion still do not appear to be added when using the variants module in a project that doesn't already have them.

I tried to activate `variants` in `prems` with these changes:

![image](https://github.com/data-team-uhn/cards/assets/651980/847292c0-ad07-46d3-8e54-9c4fcfe8ec79)

The variants ui is available but non-functional due to missing subject types:

![image](https://github.com/data-team-uhn/cards/assets/651980/334521bd-170d-4b6b-8f6e-f58733d8f041)

sdumitriu commented:
> > You are right, a feature file would be great, now that it's easy to enable them in a project
> 
> I'm encountering the same problem as before the icommit that introduces the variants feature file: Subject types Tumor, TumorRegion still do not appear to be added when using the variants module in a project that doesn't already have them.
> 
> I tried to activate `variants` in `prems` with these changes:
> 

That's not how feature files are used. You must follow the instructions in the PR description to activate the feature.

sashaandjic reviewed (approved):
tested in `prems` and `heracles` modes:
![image](https://github.com/data-team-uhn/cards/assets/23243244/7da1d691-1fef-4400-a0a6-dc6c0f99abde)
![image](https://github.com/data-team-uhn/cards/assets/23243244/74402c30-7444-439e-9673-93078178ce09)
![image](https://github.com/data-team-uhn/cards/assets/23243244/55642c0f-b895-40dd-b725-684ad0533685)
